### PR TITLE
Feat/support 4844

### PR DESCRIPTION
**Pull request by smtmfft**
- [X] Use patched ethers-rs to support blob tx.
   patched for now, and wait for upstream upgrade(seems no schedule so far)
- [X]  Read sidecar from beacon url
- [X]  Parse/decode blob data from sidecar response
- [X]  Pass tx blob list to guest.
- [x]  Update public input to support tx blob hash.
- [ ]  Check blob validness in guest

### CODE REVIEW

File: primitives/src/taiko/protocol_instance.rs [hunk 1]
```diff
@@ -103,15 +104,27 @@ impl ProtocolInstance {
 
     // keccak256(abi.encode(tran, newInstance, prover, metaHash))
     pub fn hash(&self, evidence_type: EvidenceType) -> B256 {
+        println!("chain_id: {:?}", self.chain_id);
```
Comment (johntaiko) [inline primitives/src/taiko/protocol_instance.rs:107]: These will appear in both `zkvm` and `sgx` guests, maybe need to remove

Comment (smtmfft) [reply primitives/src/taiko/protocol_instance.rs]: removed!

File: lib/src/taiko/host.rs [hunk 2]
```diff
@@ -220,14 +261,52 @@ pub fn get_taiko_initial_data<N: NetworkStrategyBundle<TxEssence = EthereumTxEss
...
     let proposeBlockCall {
-        params: _,
+        params: propose_params,
         txList: l2_tx_list,
     } = decode_propose_block_call_args(&propose_tx.input)?;
 
+    // blobUsed == (txList.length == 0) according to TaikoL1
+    let blob_used = l2_tx_list.is_empty();
+    let (l2_tx_list_blob, tx_blob_hash) = if blob_used {
+        let BlockParams {
```
Comment (johntaiko) [inline lib/src/taiko/host.rs:272]:
```suggestion
        let BlockParams { .. }
```

Comment (smtmfft) [reply lib/src/taiko/host.rs:272]: updated

**Pull request merged**